src/pointer-analysis/dereference.cpp

```diff
@@ -1911,36 +1938,24 @@ std::vector<expr2tc> dereferencet::extract_bytes_from_array(
     subtype = get_uint8_type(); //XXX signedness of chars
   }
 
-  bool is_big_endian =
-    config.ansi_c.endianess == configt::ansi_ct::IS_BIG_ENDIAN;
-
   // Calculating how many bytes are occupied by each array element
   unsigned int bytes_per_index = subtype->get_width() / 8;
-  std::vector<expr2tc> exprs(num_bytes);
+  std::vector<expr2tc> exprs;
+  exprs.reserve(num_bytes);
+
   // Calculating the array index based on the given byte offset
-  expr2tc accuml_offs =
+  expr2tc elem_idx =
     div2tc(offset->type, offset, gen_long(offset->type, bytes_per_index));
-  for(unsigned int i = 0; i < num_bytes; i++)
+  expr2tc zero = gen_zero(offset->type);
+  expr2tc elem_offs =
+    modulus2tc(offset->type, offset, gen_long(offset->type, bytes_per_index));
+  for(unsigned i = 0, n; i < num_bytes; i += n)
```

Comment:
Could you add a comment that gives more details (perhaps one-two examples) on what this loop does?

Reply:
I've added comments explaining the loop and the (somewhat hidden) assumption this function makes.